common/djangoapps/student/views.py

```diff
@@ -2559,6 +2559,7 @@ def change_email_settings(request):
     return JsonResponse({"success": True})
 
 
+@ensure_csrf_cookie
```

Comment:
Although this works on functions, I think it's a totally different story on class-based views.

I think you should use the [Django class based-views decorator](https://docs.djangoproject.com/en/1.8/topics/class-based-views/intro/#decorating-the-class):

```
from django.contrib.auth.decorators import login_required
from django.utils.decorators import method_decorator
from django.views.generic import TemplateView

class ProtectedView(TemplateView):
    template_name = 'secret.html'

    @method_decorator(login_required)
    def dispatch(self, *args, **kwargs):
        return super(ProtectedView, self).dispatch(*args, **kwargs)
```